inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/sidebar/RecommendationSidebar.java

```diff
@@ -67,5 +91,98 @@ public RecommendationSidebar(String aId, IModel<AnnotatorState> aModel,
                 aAnnotationPage.actionRefreshDocument(_target)));
         
         add(form);
-    }
+
+		chartContainer = new Label("chart-container" );
+		add(chartContainer);
+
+		add(new AbstractAjaxTimerBehavior(Duration.milliseconds(5000)) {
```

Comment:
I assume the problem you are solving with the timer behavior is to ensure that the chart regularly updates itself as the user is performing actions. But instead on relying on an unconditional timer, I'd suggest we listen for events generated in the backed an/or hook into the rendering pipeline for the main editor window.

E.g. you can add a method like this to the sidebar and this is triggered whenever the main editor view is refreshed:

```
    @OnEvent
    public void onRenderAnnotations(RenderAnnotationsEvent aEvent)
    {
       ...
    }
```

Reply:
I want to be able to append javascript to the page. With this event listener I don't get target object where I can append the javascript. Am I missing something?

Reply:
You can call `event.getRequestHandler()` which returns a `IPartialPageRequestHandler` that's about the same thing as an `AjaxRequestTarget` (i.e. `AjaxRequestTarget` implements `IPartialPageRequestHandler`).

Reply:
The difference is that `IPartialPageRequestHandler` would also be available for web-socket-based communication if we switch to that one day - `AjaxRequestTarget` is limited to AJAX. In most case, a `IPartialPageRequestHandler` is sufficient.

Reply:
Done.